src/App.tsx

```diff
@@ -13,12 +13,112 @@ const goodsFromServer: string[] = [
   'Jam',
   'Garlic',
 ];
+const numbers: number[] = Array.from(Array(10).keys());
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+const App: React.FC = () => {
+  const [isVisible, setIsVisible] = useState(true);
+  const [goods, setGoods] = useState(goodsFromServer);
+  const [goodLength, setGoodLength] = useState('1');
+
+  const reversedGoods = () => {
+    setGoods((current) => [...current].reverse());
+  };
+
+  const sortedGoods = () => {
+    setGoods((current) => [...current].sort((a, b) => a.localeCompare(b)));
+  };
+
+  const resetGoods = () => {
+    setGoods(goodsFromServer);
+  };
+
+  const sortByLengthGoods = () => {
+    setGoods((current) => [...current].sort((a, b) => a.length - b.length));
+  };
+
+  const changeLength = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    const newGoods = goodsFromServer.filter(
+      good => good.length >= Number(event.target.value),
+    );
+
+    setGoodLength(event.target.value);
+    setGoods(newGoods);
+  };
+
+  const showGoods = () => {
+    setIsVisible((current) => !current);
+  };
+
+  return (
+    <div className="App">
+      {isVisible && (
+        <button
+          className="initial-button"
+          type="button"
+          onClick={showGoods}
+        >
+          Start
+        </button>
+      )}
+      {!isVisible && (
```

Comment:
Дважды используешь лолический оператор &&, можно подправить конструкцию:

```suggestion
      {isVisible ? (
        <button
          className="initial-button"
          type="button"
          onClick={showGoods}
        >
          Start
        </button>
      ) : (
```

Reply:
Its in last version. In this without ? : 
)